UnitySDK/Assets/ML-Agents/Scripts/Sensor/VectorSensor.cs

```diff
@@ -83,6 +83,12 @@ void Clear()
 
         void AddFloatObs(float obs)
         {
+#if DEBUG
```

Comment:
All of the Agent.AddVectorObs methods wind up here.

Reply:
We could add a flag to bypass the check if the source was an int, bool, or one-hot if that's worth optimizing for...